src/Jobs/CreateStorageSymlinks.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Artisan;
+use Stancl\Tenancy\Contracts\Tenant;
+use Stancl\Tenancy\Events\CreatingStorageSymlink;
+use Stancl\Tenancy\Events\StorageSymlinkCreated;
+
+class CreateStorageSymlinks implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    public Tenant $tenant;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct(Tenant $tenant)
+    {
+        $this->tenant = $tenant;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        event(new CreatingStorageSymlink($this->tenant));
+
+        Artisan::call('tenants:link', [
+            '--tenants' => [$this->tenant->getTenantKey()],
+        ]);
```

Comment:
@lukinovec Do you think the structure should be a job calling an artisan command, or the opposite?

The opposite seems more natural to me.

Or perhaps we could create an action class to extract the logic?

Let me know which solution makes the most sense to you.

Reply:
Yeah, the opposite seems more natural. But the command has a lot of output, which makes extracting the command more difficult. How would you go about the output?

Reply:
Seems like the nature of those messages is either:
- success message
- error message

Is that correct?

We could handle stuff like the `--remove` in the **command**, based on that execute the right **action/job**, and show the right success message. Errors could be communicated via exceptions.

Reply:
When **a** link gets removed, we get a success message, and after all links get removed, we get the final success message. Same goes for creating the links. So if 5 links are created, 6 success messages get displayed

> We could handle stuff like the --remove in the command, based on that execute the right action/job, and show the right success message. Errors could be communicated via exceptions.

I don't think that would work with the multiple success messages I described

Reply:
> We could handle stuff like the --remove in the command, based on that execute the right action/job, and show the right success message. Errors could be communicated via exceptions.

https://github.com/archtechx/tenancy/pull/909/commits/2594b1960fbf663b02292c9740e11977ba313f37

I handled this the way you described with an addition to make multiple success messages work (passing a closure that outputs a message if the removal/creation was successful for each link).

@stancl, could you please review the file structure? I wasn't sure where to put the actions and the trait.

Reply:
Done